matplotcheck/base.py

```diff
@@ -929,19 +915,38 @@ def assert_xydata(
             """We use `assert_array_max_ulp()` to compare the
             two datasets because it is able to account for small errors in
             floating point numbers, and it scales nicely between extremely
-            small or large numbers. We catch this error and throw our own so
-            that we can use our own message."""
+            small or large numbers. Because of the way that matplotlib stores
+            datetime data, this is essential for comparing high-precision
+            datetime data (i.e. millisecond or lower).
+
+            We catch this error and throw our own so that we can use our own
+            message."""
             try:
                 np.testing.assert_array_max_ulp(
-                    np.array(xy_data["x"]), np.array(xy_expected[xcol])
+                    xy_data["x"].to_numpy(dtype=np.float64),
+                    xy_expected[xcol].to_numpy(dtype=np.float64),
+                    5,
                 )
             except AssertionError:
+                # xy_data and xy_expected do not contain the same data
                 raise AssertionError(message)
+            except ValueError:
+                # xy_data and xy_expected do not have the same shape
+                raise ValueError(
+                    "xy_data and xy_expected do not have the same shape"
+                )
             try:
                 np.testing.assert_array_max_ulp(
-                    np.array(xy_data["y"]), np.array(xy_expected[ycol])
+                    xy_data["y"].to_numpy(dtype=np.float64),
+                    xy_expected[ycol].to_numpy(dtype=np.float64),
+                    5,
                 )
+
             except AssertionError:
+                # xy_data and xy_expected do not contain the same data
+                raise AssertionError(message)
+            except ValueError:
+                # xy_data and xy_expected do not have the same shape
                 raise AssertionError(message)
```

Comment:
This should be 
```
raise ValueError(
                    "xy_data and xy_expected do not have the same shape"
                )
```
like the error raised above, right? Otherwise this except statement isn't very useful.

Reply:
let's go ahead and raise a valueError here!!